### PR TITLE
Account deleting

### LINKED ISSUE #13: Wyrejestrowanie

### PR DESCRIPTION
**Pull request by lychanl**
Closes #13

### CODE REVIEW

**Comment by Delebrith** [comment]
UI looks bad on bigger screens. Aesthetic value is questionable - screen seems empty, the button has black text, which is different from all the other buttons in the application. It also does not show profile picture, nor username (and I registered a user with one). I am not sure if the request to /me is actually sent. See image:
[https://tmpfiles.org/dl/46806/123.PNG](url)

The "username" label seems missused. It suggests that user is meant to use it to login. It's more of a display name.

**Comment by Delebrith** [comment]
Ok, there is a bug in AUthenticatedUserDto. Instead of "{id}" in url there should be "%d" like in UserDto. Also server url should be added to pictureUrl

File: shoppingm8_fe/lib/user/accountManagementWidget.dart [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+    if (response.statusCode == 200) {
+      setState(() {
+        _userDto = UserDto.fromJson(response.data);
+      });
+    }
+  }
+
+  ImageProvider _displayUserImage(String imageUrl) {
+    if (imageUrl != null)
+      return NetworkImage(imageUrl);
```
Comment (Delebrith) [inline shoppingm8_fe/lib/user/accountManagementWidget.dart]: You may want to add header with token to the NetworkImage.

Comment (Delebrith) [inline shoppingm8_fe/lib/user/accountManagementWidget.dart]: Please fix the url by concatenation with serverUrl

**Review by Delebrith — CHANGES_REQUESTED** [review]

**Comment by lychanl** [comment]
Unfortunately I have no idea how to visually improve layout, I am however open to suggestions.

File: shoppingm8_fe/lib/user/accountManagementWidget.dart (hunk 1 above)
Comment (lychanl) [reply shoppingm8_fe/lib/user/accountManagementWidget.dart]: It was based on userLabelWidget, but I will fix it.

Comment (lychanl) [reply shoppingm8_fe/lib/user/accountManagementWidget.dart]: As above.

**Comment by lychanl** [comment]
Changed visual style to make it more similar to login screen.

File: shoppingm8_fe/lib/user/accountManagementWidget.dart (hunk 1 above)
Comment (lychanl) [reply shoppingm8_fe/lib/user/accountManagementWidget.dart]: Changed

Comment (lychanl) [reply shoppingm8_fe/lib/user/accountManagementWidget.dart]: Changed

**Comment by Delebrith** [comment]
> 
> 
> Changed visual style to make it more similar to login screen.

Looks better. I would probably delete "display name:" label. The name itself might be sufficient; under it we may add email address, with font smaller than name and in grey color. I would place "Your account" heading in the top of the card; the picture, name and email in ~2/3 of height and move the delete button much lower. The colon after "Your account" should definetely be removed. And also, please, add more horizontal padding to the button text and change its color to white.

**Comment by lychanl** [comment]
Is it now possible to retrieve email in app, or will it be necessary to change /me endpoint?

**Comment by Delebrith** [comment]
> 
> 
> Is it now possible to retrieve email in app, or will it be necessary to change /me endpoint?

It is included in AuthenticatedUserDto there

**Review by Delebrith — APPROVED** [review]

**Pull request merged**